src/test/java/com/artipie/CredentialsFromStorageYamlTest.java

```diff
@@ -23,38 +23,145 @@
+        this.creds(new ImmutablePair<>(jane, pass), new ImmutablePair<>(john, pass));
+        MatcherAssert.assertThat(
+            new Credentials.FromStorageYaml(this.storage, this.key).users()
+                .toCompletableFuture().join(),
+            Matchers.containsInAnyOrder(jane, john)
+        );
+    }
+
     @Test
-    void readsYamlFromStorage() throws InterruptedException {
-        final Storage storage = new InMemoryStorage();
-        final Key key = new Key.From("_cred.yaml");
-        final String yaml = String.join(
-            "\n",
-            "credentials:",
-            "  jane:",
-            "    pass: plain:123",
-            "  john:",
-            "    pass: plain:abc"
+    void addsUser() {
+        final String maria = "maria";
+        final String olga = "olga";
+        final String pass = "abc";
+        this.creds(new ImmutablePair<>(maria, pass));
+        new Credentials.FromStorageYaml(this.storage, this.key).add(olga, pass)
+            .toCompletableFuture().join();
+        MatcherAssert.assertThat(
+            new PublisherAs(this.storage.value(this.key).join())
+                .asciiString().toCompletableFuture().join(),
+            new IsEqual<>(
+                this.getYaml(new ImmutablePair<>(maria, pass), new ImmutablePair<>(olga, pass))
+            )
         );
-        new BlockingStorage(storage).save(key, yaml.getBytes(StandardCharsets.UTF_8));
+    }
+
+    @Test
+    void updatesUser() {
+        final String jack = "jack";
+        final String silvia = "silvia";
+        final String old = "345";
+        final String newpass = "000";
+        this.creds(new ImmutablePair<>(jack, old), new ImmutablePair<>(silvia, old));
+        new Credentials.FromStorageYaml(this.storage, this.key).add(silvia, newpass)
+            .toCompletableFuture().join();
         MatcherAssert.assertThat(
-            new Credentials.FromStorageYaml(storage, key).users().toCompletableFuture().join(),
-            Matchers.containsInAnyOrder("jane", "john")
+            new PublisherAs(this.storage.value(this.key).join())
+                .asciiString().toCompletableFuture().join(),
+            new IsEqual<>(
+                this.getYaml(new ImmutablePair<>(jack, old), new ImmutablePair<>(silvia, newpass))
+            )
         );
     }
 
+    @Test
+    void removesUser() {
+        final String mark = "mark";
+        final String ann = "ann";
+        final String pass = "123";
+        this.creds(new ImmutablePair<>(mark, pass), new ImmutablePair<>(ann, pass));
+        new Credentials.FromStorageYaml(this.storage, this.key).remove(ann)
+            .toCompletableFuture().join();
+        MatcherAssert.assertThat(
+            new PublisherAs(this.storage.value(this.key).join())
+                .asciiString().toCompletableFuture().join(),
+            new IsEqual<>(this.getYaml(new ImmutablePair<>(mark, pass)))
+        );
+    }
+
+    @Test
+    void doNotChangeYamlIfUserNotFound() {
```

Comment:
@olenagerasimova please specify in this test method name that we are talking about removing user